### PR TITLE
support Spectrum1D reading from file with flux in counts

### PR DESCRIPTION
**Pull request by kecnry**
this PR implements an exception to support reading a Spectrum1D from a file with fluxes in counts.  This does not however address more general reading round-tripping such as the one reported in #997.  Note also that this is just hardcoding an additional exception, `Spectrum1D` currently doesn't seem to do any unit checking on the inputs for flux, and so other units might result in the same error (including non-sensical units, like fluxes with units of meters).

Example (that used to raise an `OSError: Could not identify column containing the flux`):

```
import specutils
import numpy as np
import astropy.units as u

sp = specutils.Spectrum1D(flux=np.linspace(0,1,11)*u.Unit('count'), spectral_axis=np.random.random(11)*u.Unit('J'))
sp.write("spec.fits", overwrite=True)

specutils.Spectrum1D.read("spec.fits")
```

### CODE REVIEW

**Comment by pllim** [comment]
> u.Unit('J')

You sure you didn't mean Jy?

**Comment by pllim** [comment]
Are we able to turn that example you provided into a test?

**Comment by kecnry** [comment]
> You sure you didn't mean Jy?

No, this is for spectral axis, not flux, so I just chose a random compatible unit for the spectral axis.  Swap out with Hz or um if you'd rather!

> Are we able to turn that example you provided into a test?

Sure, we could provide a regression test for this same hardcoded case, but it really would only test that someone doesn't revert this commit... so I'm wondering if it makes more sense to write general tests with the longer-term solution?  Happy to basically copy-paste this in to a test though just to make sure it doesn't raise errors.

**Comment by pllim** [comment]
> No, this is for spectral axis, not flux

Doh! Ignore me... 😆 

> but it really would only test that someone doesn't revert this commit

Right, we wouldn't want regression, which could happen if we ever refactor the internals.

File: specutils/tests/test_spectrum1d.py [hunk 1]
```diff
@@ -357,6 +357,18 @@ def test_create_with_uncertainty():
         Spectrum1D(spectral_axis=wavelengths*u.um, flux=flux, uncertainty=uncertainty)
 
 
+def test_flux_unit_io_roundtrip(tmpdir):
```
Comment (pllim) [inline specutils/tests/test_spectrum1d.py:360]:
```suggestion
def test_flux_unit_io_roundtrip(tmp_path):
```

File: specutils/tests/test_spectrum1d.py [hunk 2]
```diff
@@ -357,6 +357,18 @@ def test_create_with_uncertainty():
...
+    # regression test for https://github.com/astropy/specutils/pull/1018
+    fname = str(tmpdir.join('flux_unit_io_roundtrip.fits'))
```
Comment (pllim) [inline specutils/tests/test_spectrum1d.py:362]:
```suggestion
    fname = str(tmp_path / 'flux_unit_io_roundtrip.fits')
```

File: specutils/tests/test_spectrum1d.py (hunk 1 above)
Comment (pllim) [inline specutils/tests/test_spectrum1d.py:360]:
```suggestion
@pytest.mark.parametrize("flux_unit", ["adu", "ct/s", "count"])
def test_flux_unit_io_roundtrip(tmpdir, flux_unit):
```

File: specutils/tests/test_spectrum1d.py [hunk 3]
```diff
@@ -357,6 +357,18 @@ def test_create_with_uncertainty():
...
+    for flux_unit in ('adu', 'ct/s', 'count'):
+        sp = Spectrum1D(flux=np.linspace(0,1,11)*u.Unit(flux_unit),
+                        spectral_axis=np.random.random(11)*u.Unit('Hz'))
+        sp.write(fname, overwrite=True)
+
+        sp_load = Spectrum1D.read(fname)
+        assert sp_load.flux.unit == sp.unit
```
Comment (pllim) [inline specutils/tests/test_spectrum1d.py:369]:
```suggestion
    sp = Spectrum1D(flux=np.ones(11) * u.Unit(flux_unit),
                    spectral_axis=np.arange(1, 12) * u.Unit('Hz'))
    sp.write(fname, overwrite=True)

    sp_load = Spectrum1D.read(fname)
    assert sp_load.flux.unit == sp.flux.unit
```

**Review by pllim — COMMENTED** [review]
I think the test could be written using `parametrize` and there is no need for random number generator. Please see suggestions.

Not sure what's up with the change log, maybe @rosteen only updates it on release? https://github.com/astropy/specutils/blob/main/CHANGES.rst

**Comment by rosteen** [comment]
> Not sure what's up with the change log, maybe @rosteen only updates it on release? https://github.com/astropy/specutils/blob/main/CHANGES.rst

Yeah, I've been updating it when I do a release. I'm certainly not opposed to people updating it when they make a PR though!

I'm going to approve this, but generally agree with your suggestions @pllim . I'm about to log off for vacation but feel free to merge this once those are addressed.

**Review by rosteen — APPROVED** [review]

**Review by dhomeier — CHANGES_REQUESTED** [review]
I agree with the requested changes, otherwise this looks sensible. Yes, `Spectrum1D.__init__` does not make any checks on flux `PhysicalType`, but there you are directly passing your quantity. `parsing_utils` are searching for some compatible data fields in an input file, so it makes sense to be more conservative.

**Comment by pllim** [comment]
@dhomeier , does the PR look okay to you now, or is there something else? Thanks!

**Review by dhomeier — APPROVED** [review]

**Comment by dhomeier** [comment]
Everything you had requested as far as I can see; good to me too, thanks!

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**